charts/jaeger/templates/collector-deploy.yaml

```diff
@@ -32,6 +32,12 @@ spec:
         {{- toYaml .Values.collector.podLabels | nindent 8 }}
 {{- end }}
     spec:
+      {{- if .Values.provisionDataStore.elasticsearch }}
+      initContainers:
+        - name: elasticsearch-checker
+          image: curlimages/curl
+          command: ["sh", "-c", "until curl -s http://{{ .Release.Name }}-elasticsearch:9200/_cluster/health?wait_for_status=green; do echo waiting for Elasticsearch; sleep 5; done;"]
```

Comment:
I think you want to use `elasticsearch.client.url` from _helpers, not to compose the host name again